### PR TITLE
refactor(components): refactor breadcrumb

### PR DESCRIPTION
**Pull request by sxzz**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/tokens/breadcrumb.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import type { InjectionKey } from 'vue'
+import type { BreadcrumbProps } from '@element-plus/components/breadcrumb/src/breadcrumb'
```
Comment (JeremyWuuuuu) [inline packages/tokens/breadcrumb.ts:2]: Can we use 

```ts
import type { BreadcrumbProps } from '@element-plus/components/breadcrumb'
```
here?

Comment (JeremyWuuuuu) [reply packages/tokens/breadcrumb.ts:2]: Even though for types it won't raise error but we should be carefully handling these imports.

**Review by jw-foss — APPROVED** [review]
Left a suggestion for you to consider, feel free to merge this.

**Comment by sxzz** [comment]
Done.

File: packages/tokens/breadcrumb.ts (hunk 1 above)
Comment (sxzz) [reply packages/tokens/breadcrumb.ts:2]: If do this, the `hooks` build will failed.

**Pull request merged**